packages/grafana-ui/src/components/Toggletip/Toggletip.tsx

```diff
@@ -0,0 +1,301 @@
+  }: ToggletipProps) => {
+    const styles = useStyles2(getStyles);
+    const containerStyle = styles[theme];
+    const contentRef = useRef(null);
+    const [controlledVisible, setControlledVisible] = React.useState(false);
+
+    useEffect(() => {
+      if (controlledVisible) {
+        const handleKeyDown = (enterKey: KeyboardEvent) => {
+          if (enterKey.key === 'Escape') {
+            setControlledVisible(false);
+            if (!!onClose) {
+              onClose();
+            }
+          }
+        };
+        document.addEventListener('keydown', handleKeyDown);
+        return () => {
+          document.removeEventListener('keydown', handleKeyDown);
+        };
+      } else if (controlledVisible === false && !!onClose) {
+        onClose();
+      }
+      return;
+    }, [controlledVisible, onClose]);
+
+    const { getArrowProps, getTooltipProps, setTooltipRef, setTriggerRef, visible, update } = usePopperTooltip({
+      visible: controlledVisible,
+      placement: placement,
+      interactive: true,
+      offset: [0, 8],
+      trigger: 'click',
+      onVisibleChange: setControlledVisible,
+    });
+
+    return (
+      <>
+        {React.cloneElement(children, {
+          ref: setTriggerRef,
+          tabIndex: 0,
+        })}
+        {visible && (
+          <Portal>
+            <div
+              data-testid="toggletip-content"
+              ref={setTooltipRef}
+              {...getTooltipProps({ className: containerStyle })}
+            >
+              {!!title && (
+                <div className="toggletip-header">
+                  {typeof title === 'string' && title}
+                  {title && React.isValidElement(title) && React.cloneElement(title)}
+                </div>
+              )}
+              {closeButton && (
+                <div data-testid="toggletip-header-close" className="toggletip-header-close">
+                  <IconButton
+                    aria-label="Close Toggletip"
+                    name="times"
+                    size="md"
+                    onClick={() => setControlledVisible(false)}
+                  />
+                </div>
+              )}
+              <div ref={contentRef} {...getArrowProps({ className: 'toggletip-arrow' })} />
+              <div className="toggletip-body">
+                {typeof content === 'string' && content}
+                {React.isValidElement(content) && React.cloneElement(content)}
+                {typeof content === 'function' && update && content({ update })}
+              </div>
+              {!!footer && (
+                <div className="toggletip-footer">
+                  {typeof footer === 'string' && footer}
+                  {footer && React.isValidElement(footer) && React.cloneElement(footer)}
+                </div>
+              )}
+            </div>
+          </Portal>
+        )}
+      </>
+    );
+  }
+);
+
+Toggletip.displayName = 'Toggletip';
+
+function getStyles(theme: GrafanaTheme2) {
```

Comment:
Could we return an object instead of having one big css? At first I thought it was done like this to be able to override popper css but then I noticed that we are using the `toggletip-arrow` arrow class in the code.

Reply:
In fact is done in this way to be able to merge the react-popper-tooltip css and my css on the arrow element.

Reply:
Okay, super! I have added a suggestion on how to do this without introducing static class names so we still can leverage all the benefits of emotion.

Reply:
thank you. I like it. I already committed your suggestions.